src/ott-analytics.js

```diff
@@ -289,6 +293,9 @@ class OttAnalytics extends BasePlugin {
       position: params.position,
       playerData: playerData
     };
+    if (params.programId) {
+      bookMark.programId = params.programId;
```

Comment:
shouldn't this also come from config in case of the live channel case where app provides the program ID from its EPG data?

Reply:
that come from the epg data , look at line 255

Reply:
when playing linear live there is no EPG data, but app has the EPG data and can provide it on its own.
See the spec in the internal ticket:
![image](https://user-images.githubusercontent.com/5461862/140699678-d570704f-ff53-4c9b-9fa4-d0edf96bd667.png)

Reply:
in such case the epgId was provided from the FE , it will be at the same place in the config (this.player.sources.metadata.epgId), that is already done in the kalturar-player.js level, so from the Ott-analytics point of view, it dose not matter, either way it dose the same logic.

Reply:
no, this data is updated on the fly and app can't refresh sources data and in general it is not expected from app to do it on player level, only plugin level cause this is the only plugin that cares about this data.
The player is playing a live channel(that has programs in it) and when they change the app knows it cause it also manages EPG so it can provide EPG data dynamically

Reply:
@OrenMe the app can custom the epgid via the media options passed to loadMedia. e.g.
```
kalturaPlayer.loadMedia({entryId: ENTRY_ID}, {metadata: {egpid: CUSTOM}})
```

Reply:
@OrenMe got you but if i get it right the media (program) is changed by calling the loadMedia() and the loadmedia overdie the config every time it called (by calling to configure())